Release notes for 8.17 (#17146)
coq/coq: doc/sphinx/changes.rst

Summary of changes
~~~~~~~~~~~~~~~~~~

> Coq version 8.17 integrate a soudness fix to the Coq kernel along

Reviewer: ```suggestion
Coq version 8.17 integrates a soudness fix to the Coq kernel along
```

Reviewer 2: ```suggestion
Coq version 8.17 integrates a soundness fix to the Coq kernel along
```

with a few new features and a host of improvements to the Ltac2 language
and libraries. We highlight some of the most impactful changes here:

  - :ref:`Fixed <817VmCompute>` a logical inconsistency due to :tacn:`vm_compute` in
    presence of side-effects in the enviroment (e.g. using `Back` or `Fail`).

  - It is now possible to dynamically :ref:`enable or disable <817Notations>` notations.

>   - Support :ref:`multiple scopes <817Scopes>`` in :cmd:`Arguments` and :cmd:`Bind Scope`.

Reviewer 2: ```suggestion
  - Support :ref:`multiple scopes <817Scopes>` in :cmd:`Arguments` and :cmd:`Bind Scope`.
```

>   - The tactics chapter of the manual has seen :ref:`many improvements <817TacticsRefman>`
>     in presentation and wording and is semi-automatically checked for coherence with the implementation.

Reviewer 2: ```suggestion
  - The tactics chapter of the manual has :ref:`many improvements <817TacticsRefman>`
    in presentation and wording.  The documented grammar is semi-automatically checked
    for consistency with the implementation.
```

>   - :ref:`Fixes <817Eauto>` to the `auto` and `eauto` tactics, to respect hint priorities and the documented use

Reviewer 2: ```suggestion
  - :ref:`Fixes <817Eauto>` to the :tacn:`auto` and :tacn:`eauto` tactics, to respect hint priorities and the documented use
```

>     of :tacn:`simple apply`. Potentially breaking change.

Reviewer 2: ```suggestion
    of :tacn:`simple apply`. This is a potentially breaking change.
```

  - :ref:`New LTac2 <817Ltac2>` APIs, deep pattern-matching with ``as`` clauses and handling of literals,
    support for record types and preterms.

>   - :ref:`Move <817ClassFieldSyntax>` from :g:`:>`: to :g:`::` syntax for declaring typeclass fields as instances, fixing

Reviewer 3: ```suggestion
  - :ref:`Move <817ClassFieldSyntax>` from :g:`:>` to :g:`::` syntax for declaring typeclass fields as instances, fixing
```

    a confusion with declaration of coercions.

  - :ref:`Standard library <817Stdlib>` improvements.

See the `Changes in 8.17.0`_ section below for the detailed list of changes,
including potentially breaking changes marked with **Changed**.
Coq's `reference manual for 8.17 <https://coq.github.io/doc/v8.17/refman>`_,
`documentation of the 8.17 standard library <https://coq.github.io/doc/v8.17/stdlib>`_
and `developer documentation of the 8.17 ML API <https://coq.github.io/doc/v8.17/api>`_
are also available.

Ali Caglayan, Emilio Jesús Gallego Arias, Gaëtan Gilbert
and Théo Zimmermann worked on maintaining and improving the
continuous integration system and package building infrastructure.

Erik Martin-Dorel has maintained the `Coq Docker images
<https://hub.docker.com/r/coqorg/coq>`_ that are used in many Coq
projects for continuous integration.

Maxime Dénès, Paulo G. Giarrusso, Huỳnh Trần Khanh, and Laurent Théry have
maintained the VsCoq extension for VS Code.

> The OPAM repository for Coq packages has been maintained by

Reviewer 4: ```suggestion
The opam repository for Coq packages has been maintained by
```
The opam [website](https://opam.ocaml.org) itself uses "opam" instead of OPAM, so it may be time to change.
> opam is a source-based package manager for OCaml.

page for more details.

The 45 contributors to the 8.17 versions are:
Reynald Affeldt, Tanaka Akira, Lasse Blaauwbroek, Stephan Boyer, Ali Caglayan, Cyril Cohen, Maxime Dénès, Andrej Dudenhefner, Andres Erbsen, František Farka, Jim Fehrle, Paolo G. Giarrusso, Gaëtan Gilbert, Jason Gross, Alban Gruin, Stefan Haan, Hugo Herbelin, Wolf Honore, Bodo Igler, Jerry James, Emilio Jesús Gallego Arias, Ralf Jung, Jan-Oliver Kaiser, Wojciech Karpiel, Chantal Keller, Thomas Klausner, Olivier Laurent, Olivier Laurent and Andrej Dudenhefner, Yishuai Li, Guillaume Melquiond, Karl Palmskog, Sudha Parimala, Pierre-Marie Pédrot, Valentin Robert, Pierre Roux, Julin S, Dmitry Shachnev, Michael Soegtrop, Matthieu Sozeau, Naveen Srinivasan, Sergei Stepanenko, Karolina Surma, Enrico Tassi, Li-yao Xia
and Théo Zimmermann.

The Coq community at large helped improve this new version via
the GitHub issue and pull request system, the coq-club@inria.fr mailing list,
the `Discourse forum <https://coq.discourse.group/>`_ and the
`Coq Zulip chat <https://coq.zulipchat.com>`_.

Version 8.17's development spanned 4 months from the release of
Coq 8.17.0. Théo Zimmermann is the release manager of Coq 8.17.

> This release is the result of ?? merged PRs, closing ?? issues.

Reviewer 2: ```suggestion
This release is the result of ?? merged PRs, closing ?? issues.
```

todo

Coq's `reference manual for 8.17 <https://coq.github.io/doc/v8.17/refman>`_,
`documentation of the 8.17 standard library <https://coq.github.io/doc/v8.17/stdlib>`_
and `developer documentation of the 8.17 ML API <https://coq.github.io/doc/v8.17/api>`_
are also available.

Ali Caglayan, Emilio Jesús Gallego Arias, Gaëtan Gilbert
and Théo Zimmermann worked on maintaining and improving the
continuous integration system and package building infrastructure.

Erik Martin-Dorel has maintained the `Coq Docker images
<https://hub.docker.com/r/coqorg/coq>`_ that are used in many Coq
projects for continuous integration.

Maxime Dénès, Paulo G. Giarrusso, Huỳnh Trần Khanh, and Laurent Théry have
maintained the VsCoq extension for VS Code.

The opam repository for Coq packages has been maintained by
Guillaume Claret, Karl Palmskog, Matthieu Sozeau and Enrico Tassi with
contributions from many users. A list of packages is available at
https://coq.inria.fr/opam/www/.

The `Coq Platform <https://github.com/coq/platform>`_ has been maintained
by Michael Soegtrop, with help from Karl Palmskog, Enrico Tassi and
Théo Zimmermann.

Our current maintainers are Yves Bertot, Frédéric Besson, Ana Borges,

> Ali Caglayan, Tej Chajed, Cyril Cohen, Pierre Corbineau, Pierre Courtieu, Maxime Dénès,

Reviewer 5: We should add Andres Erbsen to the list.
And also, we should add Andres, Huỳnh and Paolo to the coq-team page (and add VsCoq to the topics of Laurent on the coq-team page).

Jim Fehrle, Julien Forest, Emilio Jesús Gallego Arias, Gaëtan Gilbert,
Georges Gonthier, Benjamin Grégoire, Jason Gross, Hugo Herbelin,
Vincent Laporte, Olivier Laurent, Assia Mahboubi, Kenji Maillard,
Guillaume Melquiond, Pierre-Marie Pédrot, Clément Pit-Claudel, Pierre Roux,
Kazuhiko Sakaguchi, Vincent Semeria, Michael Soegtrop, Arnaud Spiwack,
Matthieu Sozeau, Enrico Tassi, Laurent Théry, Anton Trunov, Li-yao Xia
and Théo Zimmermann. See the `Coq Team face book <https://coq.inria.fr/coq-team.html>`_
page for more details.

> The 45 contributors to the 8.17 versions are:

Reviewer 5: Is the "s" on purpose?
```suggestion
The 45 contributors to the 8.17 version are:
```

> Reynald Affeldt, Tanaka Akira, Lasse Blaauwbroek, Stephan Boyer, Ali Caglayan, Cyril Cohen, Maxime Dénès, Andrej Dudenhefner, Andres Erbsen, František Farka, Jim Fehrle, Paolo G. Giarrusso, Gaëtan Gilbert, Jason Gross, Alban Gruin, Stefan Haan, Hugo Herbelin, Wolf Honore, Bodo Igler, Jerry James, Emilio Jesús Gallego Arias, Ralf Jung, Jan-Oliver Kaiser, Wojciech Karpiel, Chantal Keller, Thomas Klausner, Olivier Laurent, Olivier Laurent and Andrej Dudenhefner, Yishuai Li, Guillaume Melquiond, Karl Palmskog, Sudha Parimala, Pierre-Marie Pédrot, Valentin Robert, Pierre Roux, Julin S, Dmitry Shachnev, Michael Soegtrop, Matthieu Sozeau, Naveen Srinivasan, Sergei Stepanenko, Karolina Surma, Enrico Tassi, Li-yao Xia

Reviewer 5: ```suggestion
Reynald Affeldt, Tanaka Akira, Lasse Blaauwbroek, Stephan Boyer, Ali Caglayan, Cyril Cohen, Maxime Dénès, Andrej Dudenhefner, Andres Erbsen, František Farka, Jim Fehrle, Paolo G. Giarrusso, Gaëtan Gilbert, Jason Gross, Alban Gruin, Stefan Haan, Hugo Herbelin, Wolf Honore, Bodo Igler, Jerry James, Emilio Jesús Gallego Arias, Ralf Jung, Jan-Oliver Kaiser, Wojciech Karpiel, Chantal Keller, Thomas Klausner, Olivier Laurent, Yishuai Li, Guillaume Melquiond, Karl Palmskog, Sudha Parimala, Pierre-Marie Pédrot, Valentin Robert, Pierre Roux, Julin S, Dmitry Shachnev, Michael Soegtrop, Matthieu Sozeau, Naveen Srinivasan, Sergei Stepanenko, Karolina Surma, Enrico Tassi, Li-yao Xia
```

and Théo Zimmermann.

The Coq community at large helped improve this new version via
the GitHub issue and pull request system, the coq-club@inria.fr mailing list,
the `Discourse forum <https://coq.discourse.group/>`_ and the
`Coq Zulip chat <https://coq.zulipchat.com>`_.

> Version 8.17's development spanned 4 months from the release of

Reviewer 5: We're close to 5 months now:
```suggestion
Version 8.17's development spanned 5 months from the release of
```

> Coq 8.17.0. Théo Zimmermann is the release manager of Coq 8.17.

Reviewer 5: ```suggestion
Coq 8.16.0. Théo Zimmermann is the release manager of Coq 8.17.
```

This release is the result of 414 merged PRs, closing 105 issues.

> | Nantes, January 2023,

Reviewer 5: Since the PR was not merged yet, let's update the date:

```suggestion
| Nantes, February 2023,
```
